### PR TITLE
Adds Cycle Motif Support

### PR DESCRIPTION
**Pull request by shenoynikhil**
Support for cycle motif. Based on #6255 

- [x] Adds cycle motif class
- [x] Adds test for cycle motif
- [x] Updated CHANGELOG

##### Doubt [Remove if cleared]
I have not added node classification labels (I guess these motifs don't have them)

### CODE REVIEW

File: torch_geometric/datasets/motif_generator/cycle.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    Args:
+        n (int): Number of nodes (or edges) in the cycle.
+    """
+    def __init__(self, n: int = 6):
+        # construct edge_index based on n
+        structure = Data(
+            num_nodes=n,
+            edge_index=torch.Tensor([[x for x in range(n)],
+                                     [y for y in range(1, n)] + [0]
+                                     ]).type(torch.int32),
```
Comment (wsad1) [inline torch_geometric/datasets/motif_generator/cycle.py:22]:
```suggestion
            edge_index=to_undirected(torch.stack((torch.arange(n), torch.arange(1,n+1)),
```
This will make it undirected.

File: torch_geometric/datasets/motif_generator/cycle.py [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+from torch_geometric.data import Data
+from torch_geometric.datasets.motif_generator import CustomMotif
+
+
+class CycleMotif(CustomMotif):
+    r"""Generates the cycle motif from the `"GNNExplainer:
+    Generating Explanations for Graph Neural Networks"
+    <https://arxiv.org/pdf/1903.03894.pdf>`_ paper, containing n nodes and n
+    undirected edges.
```
Comment (wsad1) [inline torch_geometric/datasets/motif_generator/cycle.py:11]:
```suggestion
    <https://arxiv.org/pdf/1903.03894.pdf>`_ paper, containing :obj:`n` nodes and
    in a cycle.
```

File: torch_geometric/datasets/motif_generator/cycle.py [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+    Args:
+        n (int): Number of nodes (or edges) in the cycle.
```
Comment (wsad1) [inline torch_geometric/datasets/motif_generator/cycle.py:14]:
```suggestion
        n (int): Number of nodes in the cycle.
```
There will be `2*n` edges since the graph is undirected.

**Review by wsad1 — COMMENTED** [review]
Looks good mostly. Left some comments.

**Review by rusty1s — APPROVED** [review]
